File: backend/routes/event_booking.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

func eventBookingRoutes(rg *gin.RouterGroup) {
	rg.GET("/event-bookings", handlers.GetEventBookings)
	rg.GET("/event-bookings/:id", handlers.GetEventBooking)
	rg.POST("/event-bookings", handlers.CreateEventBooking)
	rg.PUT("/event-bookings/:id", handlers.UpdateEventBooking)
	rg.PUT("/event-bookings/:id/submit", handlers.SubmitEventBooking)
	rg.PUT("/event-bookings/:id/approve", handlers.ApproveEventBooking)
	rg.DELETE("/event-bookings/:id", handlers.DeleteEventBooking)
}

File: frontend/src/api/event_booking.js
import http from './http'

export const getEventBookings = (params) => http.get('/event-bookings', { params })
export const getEventBooking = (id) => http.get(`/event-bookings/${id}`)
export const createEventBooking = (data) => http.post('/event-bookings', data)
export const updateEventBooking = (id, data) => http.put(`/event-bookings/${id}`, data)
export const submitEventBooking = (id, data) => http.put(`/event-bookings/${id}/submit`, data)
export const approveEventBooking = (id, data) => http.put(`/event-bookings/${id}/approve`, data)
export const deleteEventBooking = (id) => http.delete(`/event-bookings/${id}`)
